Coding/fw/lib/WebServer/Websocket.cpp

```diff
@@ -0,0 +1,131 @@
+extern "C"
+{
+}
+
+/* CONSTANTS **************************************************************************************/
+AsyncWebSocket ws("/ws");
+
+/* MACROS *****************************************************************************************/
+
+/* TYPES ******************************************************************************************/
+
+/* PROTOTYPES *************************************************************************************/
+static void handleWebSocketMessage(void *arg, uint8_t *data, size_t len);
+static void onEvent(AsyncWebSocket *server, AsyncWebSocketClient *client, AwsEventType type,
+                    void *arg, uint8_t *data, size_t len);
+
+/* VARIABLES **************************************************************************************/
+
+static QueueHandle_t inputQueue;
+
+/* PUBLIC METHODES ********************************************************************************/
+
+/**************************************************************************************************/
+
+/*
+*   Initializes the WebSocket Service
+*/
+void websocket::init(AsyncWebServer &server)
+{
+    server.on("/communication", HTTP_GET, [](AsyncWebServerRequest *request) {
+        request->send(SPIFFS, "/ws.html");
+    });
+
+    ws.onEvent(onEvent);
+    server.addHandler(&ws);
+    inputQueue = xQueueCreate(100, sizeof(char));
+    if(NULL == inputQueue)
+    {
+        Serial.println("Error at creating the Queue");
+    }
+}
+
+/**************************************************************************************************/
+
+/*
+*   Send WebSocket Message
+*/
+void websocket::send(const String &message)
+{
+    String systime = String(millis());
+    ws.textAll(message + systime);
+}
+
+/**************************************************************************************************/
+
+/*
+*   Receive WebSocket Message
+*/
+bool websocket::receive(char &c)
+{
+    return xQueueReceive(inputQueue, &c, 100);;
+}
+
+/* PROTECTED METHODES *****************************************************************************/
+
+/* PRIVATE METHODES *******************************************************************************/
+
+/* EXTERNAL FUNCTIONS *****************************************************************************/
+
+/* INTERNAL FUNCTIONS *****************************************************************************/
+
+/**************************************************************************************************/
+
+/*
+*   Handler for incoming Web Socket Message
+*/
+static void handleWebSocketMessage(void *arg, uint8_t *data, size_t len)
+{
+    String temp;
+    AwsFrameInfo *info = (AwsFrameInfo *)arg;
+    if ((info->final) && (0 == info->index) && (len == info->len) && (WS_TEXT == info->opcode))
+    {
+        for (int i = 0; i < len; i++)
+        {
+            xQueueSendToBack(inputQueue, &data[i], 100);
```

Comment:
again I would not wait for  queue full.

I also would add a debug log on queue overrun. for eample 

if (uxQueueSpacesAvailable(inputQueue) < len)
{
   Log("websockert receive overflow");
}
else
{
  /* add bytes to queue */
}